test/lib/subscriptions.js

```diff
@@ -245,158 +180,144 @@ describe("lib/subscriptions", function() {
 					new Error(`Event ${StringPermissionEvent.name} is not a registered as subscribable`)
 				);
 			});
-			it("should notify all links who subscribed to all channels", async function() {
+			it("should notify all links who subscribed", async function() {
 				subscriptions.broadcast(new RegisteredEvent());
 				await Promise.all([onceConnectorSend(0), onceConnectorSend(1)]);
 				assertLastEvent(0, RegisteredEvent); // RegisteredEvent: All
 				assertLastEvent(1, RegisteredEvent); // RegisteredEvent: All
-				assertNoEvent(2); // ChannelEvent: All
-				assertNoEvent(3); // ChannelEvent: channelOne
-				assertNoEvent(4); // ChannelEvent: channelTwo
 			});
-			it("should not notify a link who unsubscribed from all all channels", async function() {
+			it("should not notify a link who unsubscribed", async function() {
 				const connectorData = connectorSetupDate[0];
 				const request = new lib.SubscriptionRequest(RegisteredEvent.name, false);
-				await subscriptions._handleRequest(request, connectorData.src, connectorData.dst);
-				mockController.wsServer.controlConnections.get(0).connector.sentMessages.pop(); // Response to request
+				await subscriptions.handleRequest(
+					getLink(connectorData.id), request, connectorData.src, connectorData.dst,
+				);
+				getLink(0).connector.sentMessages.pop(); // Response to request
```

Comment:
This line is no longer required. Originally it was because all subscription requests gave a responce, but this is no longer true. With the new code `sentMessages` is empty at this point, and `pop` is returning `undefined` so this is effectivly a nop. Test passes with line removed.